src/teehr/visualization/dataframe_accessor.py

```diff
@@ -65,10 +79,51 @@ def _validate(obj):
 
         elif obj.attrs['table_type'] == 'location':
 
+            # check for expected fields
+            fields_list = obj.attrs['fields']
+            missing = []
+            for field in fields_list:
+                if field not in obj.columns:
+                    missing.append(field)
+            if len(missing) != 0:
+                raise AttributeError(f"""
+                    DataFrame with table_type == 'location' is missing
+                    expected column(s): {missing}
+                """)
+
+            # check for data
+            if obj.index.size == 0:
+                raise AttributeError("GeoDataFrame must have data.")
+
+            # convert to gdf if given df
+            if not (isinstance(obj, gpd.GeoDataFrame)):
+                logger.info("""
+                    Object is DataFrame. Expected GeoDataFrame. Converting to
+                    GeoDataFrame...
+                            """)
+                geo_obj = df_to_gdf(obj)
+                geo_obj.attrs['table_type'] = 'location'
```

Comment:
Correct -- attributes do not carry over when converting from pd.Dataframe to gpd.GeoDataFrame. Figured it would be best to handle this in the validation method as opposed to trying to implement it in the utils.py method. Implemented the loop you mentioned.